### PR TITLE
Introduce feature flag for enabling grouping session cache

### PR DESCRIPTION
**Pull request by bjorncs**

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/vespa/model/container/ApplicationContainerCluster.java [hunk 1]
```diff
@@ -50,11 +50,13 @@
     private ContainerModelEvaluation modelEvaluation;
 
     private Optional<TlsSecrets> tlsSecrets;
+    private final boolean enableGroupingSessionCache;
 
     public ApplicationContainerCluster(AbstractConfigProducer<?> parent, String subId, String name, DeployState deployState) {
         super(parent, subId, name, deployState);
 
         this.tlsSecrets = deployState.tlsSecrets();
+        this.enableGroupingSessionCache = deployState.getProperties().enableGroupingSessionCache();
```
Comment (gjoranv) [inline config-model/src/main/java/com/yahoo/vespa/model/container/ApplicationContainerCluster.java]: Redundant `this` (also on the line above).

File: config-model/src/main/java/com/yahoo/vespa/model/container/search/ContainerSearch.java [hunk 2]
```diff
@@ -52,6 +52,7 @@ public ContainerSearch(ApplicationContainerCluster cluster, SearchChains chains,
 
         // TODO: Should be added to container instead of cluster to get proper configId for qr config.
         cluster.addComponent(getFS4ResourcePool());
+        this.enableGroupingSessionCache = cluster.enableGroupingSessionCache();
```
Comment (gjoranv) [inline config-model/src/main/java/com/yahoo/vespa/model/container/search/ContainerSearch.java]: Redundant `this`

File: config-model/src/test/java/com/yahoo/vespa/model/container/search/test/empty.cfg [hunk 3]
```diff
@@ -0,0 +1 @@
+enableGroupingSessionCache false
```
Comment (gjoranv) [inline config-model/src/test/java/com/yahoo/vespa/model/container/search/test/empty.cfg]: Is this file necessary? If so why, and why the name "empty.cfg"?

Sigh, these cfg files in tests are a really bad idea, but not everyone agrees.

File: container-search/src/main/java/com/yahoo/search/handler/SearchHandler.java [hunk 4]
```diff
@@ -122,6 +124,7 @@ public SearchHandler(Statistics statistics,
                          ExecutionFactory executionFactory) {
         super(executor, accessLog, metric, true);
         log.log(LogLevel.DEBUG, "SearchHandler.init " + System.identityHashCode(this));
+        this.enableGroupingSessionCache = queryProfileConfig.enableGroupingSessionCache();
```
Comment (gjoranv) [inline container-search/src/main/java/com/yahoo/search/handler/SearchHandler.java]: Redundant `this`

**Review by gjoranv — APPROVED** [review]

File: config-model/src/main/java/com/yahoo/vespa/model/container/ApplicationContainerCluster.java (hunk 1 above)
Comment (bjorncs) [reply config-model/src/main/java/com/yahoo/vespa/model/container/ApplicationContainerCluster.java]: I agree that it is strictly not necessary; it's more about what code style a developer prefers.

File: config-model/src/test/java/com/yahoo/vespa/model/container/search/test/empty.cfg (hunk 3 above)
Comment (bjorncs) [reply config-model/src/test/java/com/yahoo/vespa/model/container/search/test/empty.cfg]: I have no idea 🙈 I agree, seems overkill to check in an empty file for this kind of assertion.
